packages/tools/examples/splineSegmentationROITools/index.ts

```diff
@@ -0,0 +1,466 @@
+  cornerstoneTools.addTool(PanTool);
+  cornerstoneTools.addTool(ZoomTool);
+  cornerstoneTools.addTool(StackScrollMouseWheelTool);
+  cornerstoneTools.addTool(TrackballRotateTool);
+
+  // Define tool groups to add the segmentation display tool to
+  const toolGroup = ToolGroupManager.createToolGroup(toolGroupId);
+
+  toolGroup.addTool(SegmentationDisplayTool.toolName);
+  toolGroup.addTool(SplineSegmentationROITool.toolName);
+  toolGroup.addTool(StackScrollMouseWheelTool.toolName);
+  toolGroup.addTool(PanTool.toolName);
+  toolGroup.addTool(ZoomTool.toolName);
+
+  toolGroup.addToolInstance(
+    'CatmullRomSplineROI',
+    SplineSegmentationROITool.toolName,
+    {
+      spline: {
+        type: SplineSegmentationROITool.SplineTypes.CatmullRom,
+      },
+    }
+  );
+
+  toolGroup.addToolInstance(
+    'LinearSplineROI',
+    SplineSegmentationROITool.toolName,
+    {
+      spline: {
+        type: SplineSegmentationROITool.SplineTypes.Linear,
+      },
+    }
+  );
+
+  toolGroup.addToolInstance('BSplineROI', SplineSegmentationROITool.toolName, {
+    spline: {
+      type: SplineSegmentationROITool.SplineTypes.BSpline,
+    },
+  });
+
+  toolGroup.setToolEnabled(SegmentationDisplayTool.toolName);
+
+  toolGroup.setToolActive(splineToolsNames[0], {
+    bindings: [
+      {
+        mouseButton: MouseBindings.Primary, // Left Click
+      },
+    ],
+  });
+
+  toolGroup.setToolActive(PanTool.toolName, {
+    bindings: [
+      {
+        mouseButton: MouseBindings.Auxiliary, // Middle Click
+      },
+    ],
+  });
+
+  toolGroup.setToolActive(ZoomTool.toolName, {
+    bindings: [
+      {
+        mouseButton: MouseBindings.Secondary, // Right Click
+      },
+    ],
+  });
+
+  toolGroup.setToolActive(StackScrollMouseWheelTool.toolName);
+
+  // Get Cornerstone imageIds for the source data and fetch metadata into RAM
+  const imageIds = await createImageIdsAndCacheMetaData({
+    StudyInstanceUID:
+      '1.3.6.1.4.1.14519.5.2.1.7009.2403.334240657131972136850343327463',
+    SeriesInstanceUID:
+      '1.3.6.1.4.1.14519.5.2.1.7009.2403.226151125820845824875394858561',
+    wadoRsRoot: 'https://d3t6nz73ql33tx.cloudfront.net/dicomweb',
+  });
+
+  // Instantiate a rendering engine
+  const renderingEngineId = 'myRenderingEngine';
+  const renderingEngine = new RenderingEngine(renderingEngineId);
+
+  // Create the viewports
+  const viewportInputArray = [
+    {
+      viewportId: viewportId,
+      type: ViewportType.STACK,
+      element: element,
+      defaultOptions: {
+        background: <Types.Point3>[0.2, 0, 0.2],
+      },
+    },
+  ];
+
+  renderingEngine.setViewports(viewportInputArray);
+  toolGroup.addViewport(viewportId, renderingEngineId);
+
+  // Get the stack viewport that was created
+  const stackViewport = <Types.IStackViewport>(
+    renderingEngine.getViewport(viewportId)
+  );
+
+  // Set the stack on the viewport
+  stackViewport.setStack([imageIds[0]]);
```

Comment:
Could we include two or three images to display various sections with different annotations or segments?